httpx_ws/transport.py

```diff
@@ -144,15 +147,18 @@ async def _asgi_receive(self) -> Message:
     async def _asgi_send(self, message: Message) -> None:
         self._send_queue.put(message)
 
-    def _build_accept_response(self, message: Message) -> bytes:
+    def _build_accept_response(self, message: Message) -> typing.Iterator[bytes]:
         subprotocol = message.get("subprotocol", None)
         headers = message.get("headers", [])
-        return self.connection.send(
-            wsproto.events.AcceptConnection(
-                subprotocol=subprotocol,
-                extra_headers=headers,
-            )
+        yield self.connection.send(wsproto.events.AcceptConnection(subprotocol=subprotocol, extra_headers=headers))
+
+    def _build_denial_response(
+        self, status_code: int, headers: list[tuple[bytes, bytes]], data: bytes
+    ) -> typing.Iterator[bytes]:
+        yield self.connection.send(
+            wsproto.events.RejectConnection(status_code=status_code, headers=headers, has_body=True)
         )
+        yield self.connection.send(wsproto.events.RejectData(data))
```

Comment:
The `yield` idea was bad. We can revert this.

The `_handle_ws_request` receives `accept_response` as bytes, it needs to identify that is either denial or acceptance.